sympy/core/expr.py

```diff
@@ -3440,7 +3440,16 @@ def as_leading_term(self, *symbols, logx=None, cdir=0):
 
         """
         if len(symbols) > 1:
-            c = self
+            c = self.expand()
+            if c.is_Mul:
+                return c
+            if c.is_Pow:
+                base_term = c.base.as_leading_term(*symbols, logx=logx, cdir=cdir)
+                exp_term = c.exp.as_leading_term(*symbols, logx=logx, cdir=cdir)
+                return base_term**exp_term
+            if c.is_Add:
+                symbwise_lead_terms = [c.as_leading_term(x, logx=logx, cdir=cdir) for x in symbols]
+                return c.func(*symbwise_lead_terms)
```

Comment:
Are there examples where these are needed?

Reply:
Yeah surely ! I remember correctly that @0sidharth had assisted me with this patch and its only after discussing with him for a while I had pushed the code . I'll update this message with some cases in a while ! Until then you could let me know if you have other doubts regarding the Pr.

Reply:
All blocks are needed here, I'll tell you why . So the `c.is_Add` somewhat replicates what the following block (which has been removed by us) used to do for multivariable cases . We need to somehow split and Add subparts together as was done below 
```
 if expr.is_Add:
           lst = expr.extract_leading_order(args)
           expr = Add(*[f.expr for (e, f) in lst])
```
If we don't have these blocks and stick to the original code which is written below
```
self = c
for x in symbols:
      c = c.as_leading_term(x, logx=logx, cdir=cdir)
      return c
```
we won't be getting desired results for multivariable cases as the form in which self is passed here isn't split up already . For example on master something like `O(x + y) ` or O(1/x**2 + 2*y) would always enter the `as_leading_term()` code either as` x` or `y` but not `x+y` , but now that we have removed the above code , we need to do that split up in `as_leading_term()` . I am not the best while explaining but I hope you understand !

I just added a print statement(print(self, 'this is self') as the first line in `as_leading_term()` to show what I meant and we get the following

On Master
![image](https://user-images.githubusercontent.com/87052487/140876446-f28bdb99-9fcd-4d28-93b2-ed14f4ff1e36.png)
On Branch
![image](https://user-images.githubusercontent.com/87052487/140876527-b8e912d1-4e32-41d5-ba01-a92e2be77e45.png)